enterprise/frontend/src/embedding-sdk/cli/run.ts

```diff
@@ -44,6 +47,9 @@ export const CLI_STEPS = [
     executeStep: generateReactComponentFiles,
   },
   { id: "setupLicense", executeStep: setupLicense },
+  { id: "setupEmbeddingSettings", executeStep: setupEmbeddingSettings },
+  { id: "askForTenancyColumns", executeStep: askForTenancyColumns },
+  { id: "setupPermissions", executeStep: setupPermissions },
```

Comment:
The subsequent steps continue to work even when there is no license. We have an upcoming PR to explicitly allow skipping license keys.

Reply:
I wonder if we could use a state machine or something nice for this - if the steps are going to become more complex than a linear process then we should probably come up with something that's a little more robust

Reply:
@oisincoveney I'm thinking about either `runIf: state => state.token` that makes it obvious that this step only executes if the condition is met (so the reader of the code is aware that it depends on the license step prior to this), or we explicitly add `requires: "setupLicense"` and tracks which step has been run. imo, I prefer the first one (`runIf`) over `requires`

I can add this in the license improvement PR https://github.com/metabase/metabase/pull/47229, as it's not possible to skip the license yet in this PR, but it'll be in https://github.com/metabase/metabase/pull/47229

Reply:
perfect 😄

Reply:
I think the first one makes more sense. I'd rather base the steps on the actual data rather than create a tree that we have to reason through in our heads just to understand what is going on

Reply:
@oisincoveney implemented in [this commit](https://github.com/metabase/metabase/pull/47229/commits/acfaaab181afb0dd40e7e6047e818e55d3c5479e), in the improvement PR, thanks!